app/services/correspondence_auto_assign_logger.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+class CorrespondenceAutoAssignLogger
+  def initialize
+    @total_attempts = 0
+    @failed_attempts_count = 0
+    @failed_assignments = []
+    @successful_assignments = []
+  end
+
+  def begin_logging
+    create_batch_assignment
+  end
+
+  def end_logging
```

Comment:
Let's add a parameter here to indicate why this was called so then when you incorporate table usage it becomes the value that goes into the status column
```suggestion
  def end_logging(status)
```

Reply:
Added status parameter